c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1439 @@
+        //! Router diagnostic counter event has occurred
+        uint router_diagnostic : 1;
+        //! Router packet dumped - indicates failed delivery
+        uint router_dump : 1;
+        //! Router error - packet parity, framing, or time stamp error
+        uint router_error : 1;
+        //! System Controller interrupt bit set for this processor
+        uint cpu : 1;
+        uint ethernet_tx : 1;   //!< Ethernet transmit frame interrupt
+        uint ethernet_rx : 1;   //!< Ethernet receive frame interrupt
+        uint ethernet_phy : 1;  //!< Ethernet PHY/external interrupt
+        //! System-wide slow (nominally 32 KHz) timer interrupt
+        uint slow_clock : 1;
+        //! Comms controller can accept new Tx packet
+        uint cc_tx_not_full : 1;
+        uint cc_rx_mc : 1;      //!< Comms controller multicast packet received
+        //! Comms controller point-to-point packet received
+        uint cc_rx_p2p : 1;
+        //! Comms controller nearest neighbour packet received
+        uint cc_rx_nn : 1;
+        uint cc_rx_fr : 1;      //!< Comms controller fixed route packet received
+        uint int0 : 1;          //!< External interrupt request 0
+        uint int1 : 1;          //!< External interrupt request 1
+        uint gpio8 : 1;         //!< Signal on GPIO[8]
+        uint gpio9 : 1;         //!< Signal on GPIO[9]
+    };
+    uint value;
+} vic_mask_t;
+
+//! VIC registers
+typedef struct {
+    const vic_mask_t irq_status;    //!< IRQ status register
+    const vic_mask_t fiq_status;    //!< FIQ status register
+    const vic_mask_t raw_status;    //!< raw interrupt status register
+    vic_mask_t int_select;          //!< interrupt select register
+    vic_mask_t int_enable;          //!< interrupt enable set register
+    vic_mask_t int_disable;         //!< interrupt enable clear register
+    vic_mask_t soft_int_enable;     //!< soft interrupt set register
+    vic_mask_t soft_int_disable;    //!< soft interrupt clear register
+    bool protection;                //!< protection register
+    const uint _padding[3];
+    //! current vector address register
+    vic_interrupt_handler_t vector_address;
+    //! default vector address register
+    vic_interrupt_handler_t default_vector_address;
+} vic_control_t;
+
+//! VIC individual vector control
+typedef struct {
+    uint source : 5;                //!< interrupt source
+    uint enable : 1;                //!< interrupt enable
+    uint : 26;
+} vic_vector_control_t;
+
+ASSERT_WORD_SIZED(vic_mask_t);
+ASSERT_WORD_SIZED(vic_interrupt_handler_t);
+ASSERT_WORD_SIZED(vic_vector_control_t);
+
+//! VIC registers
+static volatile vic_control_t *const vic_control =
+        (vic_control_t *) VIC_BASE_UNBUF; // NB unbuffered!
+//! VIC interrupt handlers. Array of 32 elements.
+static volatile vic_interrupt_handler_t *const vic_interrupt_vector =
+        (vic_interrupt_handler_t *) (VIC_BASE + 0x100);
```

Comment:
whats this 0x100 come from?

Reply:
`VIC_ADDR` in `spinnaker.h` (i.e., from the datasheet) except that value is broke given what we're doing it. Using the memory offset from the datasheet directly works better.

I don't know why the VIC uses such a large offset.

Reply:
(I can probably guess the immediate reason, but why they — ARM — had to use such smartass trickery in the address decode logic in the VIC I don't know. It is what it is.)